api/interservice/authz/v2/project.proto

```diff
@@ -92,19 +91,11 @@ message ProjectUpdateStatusResp {
     string failure_message = 5;
 }
 
-message ListProjectRulesReq {}
-message ProjectCollectionRulesResp {
+message ListRulesForAllProjectsReq {}
+message ListRulesForAllProjectsResp {
     map<string, ProjectRules> project_rules = 2;
 }
 
-message GetProjectRulesReq {
-    string project_id = 1;
-}
-
-message GetProjectRulesResp {
-    ProjectRules rules_for_project = 2;
-}
-
 message ProjectRules {
     repeated ProjectRule rules = 2;
 }
```

Comment:
Later in this file, it defines these:
```
enum ProjectRuleTypes {
    NODE = 0;
    EVENT = 1;
}
```
But in components/automate-gateway/api/iam/v2beta/common/rules.proto there are these:
```
enum RuleType {
    RULE_TYPE_UNSET = 0;
    NODE = 1;
    EVENT = 2;
}
```
Two concerns:
1. NODE type is a zero-value in the former, so it is suppressed in the output (as you noted in your preamble). Having the explicit unset/zero value, as the latter does, would improve the output, always showing "node" and "event".
2. Perhaps more significantly, these two definitions are inconsistent for no particular reason. Would be much better if they were consistent.

Reply:
tbh I'm not sure why they were defined that way. Looks like we're using the UNSET value [here](https://github.com/chef/automate/blob/master/components/automate-gateway/handler/iam/v2beta/rules/rules.go#L167) to do some validation at the gateway level.

In any case, I'd like to discuss these APIs and come to an agreement on what they should look like as a team.

Reply:
I'll just add that I particularly don't like that if the rule is "node" type then it does not include the type in the output changing the enum values to 1 and 2 would fix that.

Reply:
It's a good practice for GRPC APIs, I believe: due to the way that an unset enum _is its first value_. When using `grpc-gateway`, as we do, it means the user _has to provide_ the rule type -- if they don't, it'll fail as unset; without the first value, it would default to the first value. At least in my eyes, that would be a bit surprising.

Random googled up blog post re: protobuf enums: http://androiddevblog.com/protocol-buffers-pitfall-adding-enum-values/